zenodo/modules/pages/utils.py

```diff
@@ -66,6 +70,31 @@ def format_request_email_body(context):
     template = current_app.config['PAGES_EMAIL_BODY_TEMPLATE'],
     return render_template_to_string(template, context)
 
+
+def check_attachment_size(attachments):
+    """Function to check size of attachment within limits.
+
+    :param attachments: Attachments files.
+    :returns: Boolean.
+    """
+    if len(attachments) == 0:
+        return False
+    if request.content_length:
+        file_size = int(request.content_length)
+        if file_size > current_app.config['PAGES_ATTACHMENT_MAX_SIZE']:
+            return False
+    else:
+        size = 0
+        for upload in attachments:
+            upload.seek(current_app.config['PAGES_ATTACHMENT_MAX_SIZE'] -
+                        size)
+            if upload.read(1) == '':
+                return False
+
+            size += len(upload)
+    return True
+
+
 def send_support_email(context):
```

Comment:
You should get the mail instance through the registered extension of `current_app`, as done [here](https://github.com/inveniosoftware/invenio-mail/blob/master/invenio_mail/tasks.py#L47)